src/main/java/org/truffleruby/language/supercall/ReadZSuperArgumentsNode.java

```diff
@@ -70,6 +71,15 @@ public final Object execute(VirtualFrame frame) {
             superArguments = splattedArguments;
         }
 
+        // If there is an empty hash, remove it, because no keywords were passed
```

Comment:
I don't think this is correct if the KeywordDescriptor is EmptyKeywordDescriptor. But maybe it's never the case?

I think this should anyway be handled consistently on the caller or the callee side, currently it seems mostly on the callee side except this place.